encoding/rs/icicle/extend_poly.go

```diff
@@ -0,0 +1,65 @@
+//go:build icicle
+
+package icicle
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/Layr-Labs/eigenda/encoding/icicle"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/core"
+	iciclebn254 "github.com/ingonyama-zk/icicle/v3/wrappers/golang/curves/bn254"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/curves/bn254/ntt"
+	icicleRuntime "github.com/ingonyama-zk/icicle/v3/wrappers/golang/runtime"
+)
+
+type RsIcicleBackend struct {
+	NttCfg  core.NTTConfig[[iciclebn254.SCALAR_LIMBS]uint32]
+	Device  icicleRuntime.Device
+	GpuLock sync.Mutex
+}
+
+// Encoding Reed Solomon using FFT
+func (g *RsIcicleBackend) ExtendPolyEval(coeffs []fr.Element) ([]fr.Element, error) {
+	// Lock the GPU for operations
+	g.GpuLock.Lock()
```

Comment:
If it's not using GPU (which is possible for icicle), then wouldn't this lock hurt performance?

Reply:
The perf is already quite poor on the CPU version of icicle. I would prefer to tackle that separately if that's okay.